models/remoteworkitem_test.go

```diff
@@ -0,0 +1,92 @@
+// +build unit
+
+package models
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"testing"
+)
+
+func TestMap(t *testing.T) {
+
+	// This is the Type to be converted to
+	workItemType := wellKnown["1"] // Will remove this and add an actual Type.
+
+	remoteWorkItem := RemoteWorkItem{
+		Fields: map[string]interface{}{
+			"id":          "3",
+			"description": "I am a remote issue's description in Github",
+			"title":       "I am the title of the issue",
+		},
+	}
+
+	workItemMap := WorkItemMap{
+		"id":          "remote_issue_id",
+		"description": "description",
+		"title":       "title",
+	}
+	t.Log("Test remoteWorkItem", remoteWorkItem)
+	t.Log("Test workItemMap ", workItemMap)
+
+	// TODO: Add error handling
+	workItem := remoteWorkItem.MapRemote(workItemMap, workItemType)
+
+	// TODO: compare the workItem fields with those of the remoteWorkItem
+
+	t.Log("Test workItem ", workItem)
+}
+
+func TestMapGithub(t *testing.T) {
+	url := "https://api.github.com/repos/almighty/almighty-core/issues/131"
+	response, err := http.Get(url)
+
+	if err != nil {
+		t.Error("Error fetching github api data ")
+	}
+
+	defer response.Body.Close()
+	responseData, err := ioutil.ReadAll(response.Body)
```

Comment:
If you use https://golang.org/pkg/encoding/json/#NewDecoder you don't have to read the response data as a string first.